src/scripts/main.js

```diff
@@ -1,7 +1,367 @@
+
+  for (let n = 0; n < state.length; n++) {
+    for (let i = 0; i < state.length; i++) {
+      for (let j = 1; j < state[i].length; j++) {
+        if (
+          state[i][j - 1] === state[i][j] &&
+          state[i][j] !== 0 &&
+          iWasUsed.every((ind) => ind !== i)
+        ) {
+          if (
+            state[i][0] !== 0 &&
+            state[i][3] !== 0 &&
+            state[i][0] === state[i][1] &&
+            state[i][2] === state[i][3]
+          ) {
+            state[i][0] = state[i][2] * 2;
+            state[i][1] = state[i][3] * 2;
+            state[i][2] = 0;
+            state[i][3] = 0;
+            isSomethingChanged = true;
+            iWasUsed.push(i);
+            filledCells -= 2;
+            continue;
+          }
+          state[i][j - 1] *= 2;
+          state[i][j] = 0;
+          isSomethingChanged = true;
+          iWasUsed.push(i);
+          filledCells--;
+        }
+
+        if (state[i][j - 1] === 0 && state[i][j] !== 0) {
+          [state[i][j - 1], state[i][j]] = [state[i][j], state[i][j - 1]];
+          isSomethingChanged = true;
+        }
+      }
+    }
+  }
+
+  if (isSomethingChanged) {
+    addRandomNumInCells();
+  }
+
+  if ([...cells].some((cell) => cell.textContent === '2048')) {
+    messageWin.className = 'message message-win';
+  }
+
+  if (filledCells === cells.length && !canGameContinue()) {
+    messageLose.className = 'message message-lose';
+  }
+};
+
+game.moveRight = () => {
+  let isSomethingChanged = false;
+  const iWasUsed = [];
+
+  for (let n = 0; n < state.length; n++) {
+    for (let i = 0; i < state.length; i++) {
+      for (let j = state[i].length - 2; j >= 0; j--) {
+        if (
+          state[i][j + 1] === state[i][j] &&
+          state[i][j] !== 0 &&
+          iWasUsed.every((ind) => ind !== i)
+        ) {
+          if (
+            state[i][0] !== 0 &&
+            state[i][3] !== 0 &&
+            state[i][0] === state[i][1] &&
+            state[i][2] === state[i][3]
+          ) {
+            state[i][2] = state[i][0] * 2;
+            state[i][3] = state[i][1] * 2;
+            state[i][0] = 0;
+            state[i][1] = 0;
+            isSomethingChanged = true;
+            iWasUsed.push(i);
+            filledCells -= 2;
+            continue;
+          }
+          state[i][j + 1] *= 2;
+          state[i][j] = 0;
+          isSomethingChanged = true;
+          iWasUsed.push(i);
+          filledCells--;
+        }
+
+        if (state[i][j + 1] === 0 && state[i][j] !== 0) {
+          [state[i][j + 1], state[i][j]] = [state[i][j], state[i][j + 1]];
+          isSomethingChanged = true;
+        }
+      }
+    }
+  }
+
+  if (isSomethingChanged) {
+    addRandomNumInCells();
+  }
+
+  if ([...cells].some((cell) => cell.textContent === '2048')) {
+    messageWin.className = 'message message-win';
+  }
+
+  if (filledCells === cells.length && !canGameContinue()) {
+    messageLose.className = 'message message-lose';
+  }
```

Comment:
you can move these lines of code after switch/case
<img width="616" alt="image" src="https://github.com/user-attachments/assets/57d0f40d-4592-47c5-aa62-a30434d5049a">
 and write it only one time

Reply:
```
if (isSomethingChanged) {
    addRandomNumInCells();
  }
```
I can't move this line of code after switch/case, because it's a local variable for every move method. It's probably possible, but maybe it's a bit of extra work and it's not worth it. However, I have moved the other lines of code mentioned.